### PR TITLE
Rust/Ivy Module Docs

### PR DESCRIPTION
**Pull request by fosskers**
- [x] It targets the develop branch
- [x] I've searched for similar pull requests and found nothing
- [x] This change is NOT in Doom's do-not-PR list: https://doomemacs.org/d/do-not-pr
- [x] If I've bumped any packages, I've done so according to https://doomemacs.org/d/how2bump
- [x] I've linked any relevant issues and PRs below
- [x] All my commit messages are descriptive and distinct

This PR tweaks the Rust installation instructions and fixes a few minor typos.

### CODE REVIEW

File: modules/lang/rust/README.org [hunk 1]
```diff
@@ -72,19 +100,19 @@ Enable the [[file:../../../modules/editor/format/README.org][:editor format]] mo
...
+| ~<localleader> b b~ | ~cargo build~      |
+| ~<localleader> b B~ | ~cargo bench~      |
+| ~<localleader> b c~ | ~cargo check~      |
+| ~<localleader> b C~ | ~cargo clippy~     |
+| ~<localleader> b d~ | ~cargo doc~        |
+| ~<localleader> b n~ | ~cargo update~     |
+| ~<localleader> b o~ | ~cargo outdated~   |
+| ~<localleader> b r~ | ~cargo run~        |
+| ~<localleader> t a~ | ~cargo test~       |
+| ~<localleader> t t~ | ~run current test~ |
```
Comment (fosskers) [inline modules/lang/rust/README.org:115]: I swore I trashed this change. Fixing...

File: modules/completion/ivy/README.org [hunk 2]
```diff
@@ -76,7 +76,7 @@ brew install ripgrep
 
 *** Arch Linux
 #+BEGIN_SRC sh :dir /sudo::
-sudo pacman --needed --noconfirm -S ripgrep
+sudo pacman -S ripgrep
```
Comment (fosskers) [inline modules/completion/ivy/README.org:79]: While `--needed` and `--noconfirm` work just fine, it's a bit unusual to use them when installing a single package.

File: modules/lang/rust/README.org [hunk 3]
```diff
@@ -37,14 +41,38 @@ e.g. ~cargo~.
...
-~curl https://sh.rustup.rs -sSf | sh~
+** Install
+
+This module requires ~rust~, which can be acquired through =rustup=.
+
+*** Arch Linux
+
+#+begin_src sh
+sudo pacman -S rustup
+#+end_src
```
Comment (gagbo) [inline modules/lang/rust/README.org:58]: Installing a package manager through your package manager is almost always a bad idea. What makes it better with rustup ? Why would that be better than just using the maintained upstream script ?

Comment (fosskers) [reply modules/lang/rust/README.org:58]: It is general Arch Linux ethos to register as much as possible through the package manager: https://wiki.archlinux.org/title/Rust#Rustup

Comment (iyefrat) [reply modules/lang/rust/README.org:58]: I don't have experience on the rust side, but trying to install `ghcup` through `pacman` does not exactly lead to good things.

Comment (fosskers) [reply modules/lang/rust/README.org:58]: We will have to take that into account when the Haskell Foundation does their planned Unified Installer overhaul.

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Thanks for the help!

**Pull request merged**